sql/src/main/java/io/crate/expression/eval/EvaluatingNormalizer.java

```diff
@@ -167,10 +168,10 @@ public Symbol visitWindowFunction(WindowFunction function, TransactionContext co
             Function normalizedFunction = (Function) normalizeFunction(function, context);
             WindowDefinition windowDefinition = function.windowDefinition();
             OrderBy windowOrderBy = windowDefinition.orderBy();
-            if (windowOrderBy != null) {
+            if (!windowDefinition.partitions().isEmpty() || windowOrderBy != null) {
                 windowDefinition = new WindowDefinition(
```

Comment:
Would it make sense to add a `map(Function<I, O> mapper)` method to the `WindowDefinition`? Could be used here and in the `FunctionCopyVisitor`